### ISSUE TITLE
PS4 AccountID Online

### ISSUE THREAD

**Issue by CSystem77** [thestr4ng3r/chiaki#158]
If you dont have Python3 :

https://trinket.io/embed/python3/52183a157e?outputOnly=true&runOption=run&start=result&showInstructions=true

Best Regards,
CSystem

**Comment by FearlessSpiff** [comment]
Nice!

**Comment by CSystem77** [comment]
Version Web :
http://178.238.225.36/

Credit :
Script : thestr4ng3r & grill2010
Version Web : CSystem

Other :
Link Script Python : Here
Theme : jekyll theme

**Comment by thestr4ng3r** [comment]
Very nice!
But please everybody be aware that by using these online-services, you are sending your PSN Authentication Token to this website, which means you have to trust that the code running there is also exactly what is available at https://github.com/CSystem77/Account-PSN-ID and the credentials are not saved or abused. So use it at your own risk.

@CSystem77 Since you are already using php, you might want to take a look at https://github.com/Tustin/psn-php With this, I was able to determine psn ids given only the username without authenticating it. It requires an API key though.

**Comment by CSystem77** [comment]
thank you.
I would have to find a way to see the code executed and the execution link.
I can give you access to my server for you to look at the code.
I'm going to look at Tustin sources, Thank.

**Comment by thestr4ng3r** [comment]
It's fine, I only want users to be aware of potential risks of what they are doing.
Please also check the issue I created on your repository.

**Comment by grill2010** [comment]
I've also created a C# application a few months ago which is able to get the account id for every user just by the online-id (PSN user name) once you are logged in and have the authentication token. It has a nice login GUI in the application itself so no need to copy and paste URIs. It's based on [https://github.com/Tustin/PlayStationSharp](https://github.com/Tustin/PlayStationSharp). I have plans to publish it on GitHub as well. Can then be ported to other languages as well.

**Comment by thestr4ng3r** [comment]
I actually also implemented the whole OAuth process as a self-contained html page using only local JavaScript and it worked quite well, without having to copy anything manually too. The only problem there was that because of the CORS headers of the PSN api, all common browsers reject the API requests from my page without specific (complicated and browser-dependent) configuration so it turned out to be less convenient than the python script and thus useless.

**Comment by AsciiWolf** [comment]
It would be great if the functionality to get PS4 AccountID could be added to Chiaki itself if possible. The current solution is suitable only for experienced users, not regular ones.

**Comment by thestr4ng3r** [comment]
Yes, that would be the most obvious approach. Initially, I did not add it into the applications themselves on purpose since the id-fetching is conceptually different from streaming itself in the sense that it accesses live services run by Sony as opposed to only communicating locally with the console. The PSN might change at any time, so this is could be easily broken, while streaming will always work if the console stays the same and gets no updates installed.
But nowadays I think it will just be nicer for convenience to have it in Chiaki itself, especially on Android.

**Comment by DarkDevvv** [comment]
i get a error when getting token
